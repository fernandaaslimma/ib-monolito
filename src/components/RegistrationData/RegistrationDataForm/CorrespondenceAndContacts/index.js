import CorrespondenceAndContacts from "./CorrespondenceAndContacts";

export default CorrespondenceAndContacts;
