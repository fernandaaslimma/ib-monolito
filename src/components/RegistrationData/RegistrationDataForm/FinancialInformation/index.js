import FinancialInformation from "./FinancialInformation";

export default FinancialInformation;
