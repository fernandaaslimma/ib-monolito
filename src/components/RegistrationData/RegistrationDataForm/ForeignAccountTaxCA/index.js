import ForeignAccountTaxCA from "./ForeignAccountTaxCA";

export default ForeignAccountTaxCA;
