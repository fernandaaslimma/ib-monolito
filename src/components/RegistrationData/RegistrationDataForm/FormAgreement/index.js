import FormAgreement from "./FormAgreement";

export default FormAgreement;
