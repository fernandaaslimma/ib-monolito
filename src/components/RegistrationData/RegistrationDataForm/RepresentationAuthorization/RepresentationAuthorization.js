import React, { useContext } from "react";
import { IncrementalInput, AlertMessage } from "react-bocombbm-components";
import { translate } from "../../../../utils/i18n";
import { validateInputByType } from "../../../../utils/validations/input";
import deepClone from "../../../../utils/deepClone";
import {
  GridWrapper,
  Block,
  InnerGridWrapper,
  InnerBlock,
  BasicInfo,
  InnerTitle,
  Item
} from "../styles";
import Radio from "../../../common/Radio";
import { ATTORNEYS_IN_FACT } from "../../../../utils/constants";
import SectionedFormContent from "../../../common/SectionedFormContent";
import { StateContext } from "../RegistrationDataForm";

function RepresentationAuthorization({
  onChange,
  originalData,
  validateSection,
  disabled
}) {
  const { temporaryInvalidFields, isFilled, formModel, closeAll } = useContext(
    StateContext
  );
  const invalidFields = temporaryInvalidFields;
  let { representationAuthorization, attorneysInFact } = formModel.content;

  const handleChangeRadio = e => {
    representationAuthorization = e.target.value === "true";

    if (!representationAuthorization) {
      attorneysInFact = deepClone(originalData);
    }
    onChange(representationAuthorization, "representationAuthorization");
    onChange(attorneysInFact, ATTORNEYS_IN_FACT);
  };

  const filledText = representationAuthorization
    ? translate("ATUCAD_I_HAVE")
    : translate("ATUCAD_I_DO_NOT_HAVE");

  const changeIncremental = value => {
    attorneysInFact = value;
    onChange(attorneysInFact, ATTORNEYS_IN_FACT);
  };

  const validate = (item, modelIndex) => {
    const value = attorneysInFact[modelIndex][item];
    return validateInputByType[item](value);
  };

  const checkIfHasItemAndValidate = (item, modelIndex) =>
    attorneysInFact[modelIndex] &&
    attorneysInFact[modelIndex][item] &&
    validate(item, modelIndex);

  const renderInvalidMessage = () =>
    invalidFields.attorneysInFact.map(item => (
      <Item data-last={translate("ATUCAD_AND")} key={item}>
        {translate(`ATUCAD_${item.toUpperCase()}`)}
      </Item>
    ));

  return (
    <SectionedFormContent
      openedDescription={translate(
        "ATUCAD_REPRESENTATION_AUTHORIZATION"
      ).toUpperCase()}
      dataTest="SectionedRepresentationAuthorization"
      closedDescription={`${filledText} ${translate(
        "ATUCAD_ATTORNEYS_SENTENCE"
      )}`}
      valid={representationAuthorization ? isFilled.attorneysInFact : true}
      close={closeAll}
      callback={() => validateSection(attorneysInFact, ATTORNEYS_IN_FACT)}
    >
      <GridWrapper>
        {invalidFields && invalidFields.attorneysInFact && (
          <Block span={[1, 3]}>
            <AlertMessage type="error" icon="Attention">
              {translate("ATUCAD_REQUIRED_FIELDS_OMMITED")}:<br />
              {renderInvalidMessage()}
            </AlertMessage>
          </Block>
        )}
        <Block span={[1, 3]}>
          <InnerGridWrapper columns={2}>
            <InnerBlock IEFractionParts={2}>
              <Radio
                dataTest="RadionIHaveRepresentation"
                label={translate("ATUCAD_I_HAVE")}
                onChange={handleChangeRadio}
                name="authorization"
                value={true}
                checked={representationAuthorization}
                disabled={disabled}
              />
            </InnerBlock>
            <InnerBlock IEFractionParts={2}>
              <Radio
                dataTest="RadionIDoNotHaveRepresentation"
                label={translate("ATUCAD_I_DO_NOT_HAVE")}
                onChange={handleChangeRadio}
                name="authorization"
                value={false}
                checked={!representationAuthorization}
                disabled={disabled}
              />
            </InnerBlock>
            <InnerBlock span={[1, 2]} IEFractionParts={2}>
              <BasicInfo small>
                <b>____________________________</b>
                {` ${translate("ATUCAD_ATTORNEYS_SENTENCE")}`}
              </BasicInfo>
            </InnerBlock>
          </InnerGridWrapper>
        </Block>
        {representationAuthorization && (
          <Block span={[1, 3]}>
            <InnerGridWrapper>
              <InnerBlock span={[1, 3]}>
                <InnerTitle>
                  {translate("ATUCAD_ATTORNEYS_IN_FACT").toUpperCase()}
                </InnerTitle>
              </InnerBlock>
              <InnerBlock span={[1, 3]}>
                <IncrementalInput
                  inputs={[
                    {
                      type: "text",
                      name: "name",
                      label: translate("ATUCAD_NAME_INPUT_LABEL"),
                      required: true,
                      valid: checkIfHasItemAndValidate
                    },
                    {
                      type: "text",
                      name: "cpf",
                      maskType: "cpf",
                      label: translate("ATUCAD_CPF"),
                      required: true,
                      valid: checkIfHasItemAndValidate
                    }
                  ]}
                  dataTest="attorneysInFact"
                  data={attorneysInFact}
                  incrementText={translate("ADD_ONE_MORE_REPRESENTATION")}
                  spacing={{
                    left: "none",
                    top: "none",
                    right: "none",
                    bottom: "none"
                  }}
                  onChange={changeIncremental}
                  disabled={disabled}
                />
              </InnerBlock>
            </InnerGridWrapper>
          </Block>
        )}
      </GridWrapper>
    </SectionedFormContent>
  );
}

export default RepresentationAuthorization;
