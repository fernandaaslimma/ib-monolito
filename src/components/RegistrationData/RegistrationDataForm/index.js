import { connect } from "redux-zero/react";
import { combineActions } from "redux-zero/utils";

import registrationActions from "../../../actions/registration";
import notificationActions from "../../../actions/notification";
import modalActions from "../../../actions/modal";
import mfaActions from "../../../actions/mfa";
import toastrActions from "../../../actions/toastr";

import RegistrationDataForm from "./RegistrationDataForm";

export default connect(
  ({
    registrationFormData,
    dataPersonalRegistration,
    notification,
    loading,
    mfaTokenValidated,
    mfaTokenParams,
    countries,
    userInfo,
    methodChanging
  }) => ({
    registrationFormData,
    dataPersonalRegistration,
    notification,
    loading,
    mfaTokenValidated,
    mfaTokenParams,
    countries,
    userInfo,
    methodChanging
  }),
  combineActions(
    registrationActions,
    modalActions,
    notificationActions,
    mfaActions,
    toastrActions
  )
)(RegistrationDataForm);
