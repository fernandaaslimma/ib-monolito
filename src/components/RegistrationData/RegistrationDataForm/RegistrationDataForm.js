import React, { Component, Fragment } from "react";
import { func, object } from "prop-types";
import {
  Wrapper,
  InnerWrapContent,
  UserInformation,
  Greeting,
  Warning,
  BasicInfo,
  Understatement,
  TokenWrapper,
  WrapperAlert,
  TextBold,
  WrapperText
} from "./styles";

import { translate } from "../../../utils/i18n";
import deepClone from "../../../utils/deepClone";
import RegistrationDataFormResult from "../RegistrationDataFormResult";

import FinancialInformation from "./FinancialInformation";
//import ForeignAccountTaxCA from "./ForeignAccountTaxCA";
//import PurposeWithInstitution from "./PurposeWithInstitution";
import CorrespondenceAndContacts from "./CorrespondenceAndContacts";
import PersonalDetails from "./PersonalDetails";
//import RepresentationAuthorization from "./RepresentationAuthorization";
//import ProfessionalInfo from "./ProfessionalInfo";
//import OtherNationality from "./OtherNationality";
//import RelatedPerson from "./RelatedPerson";
//import PoliticallyExposedPerson from "./PoliticallyExposedPerson";

import Header from "../../common/Modal/Header";
import { MODAL_TYPES } from "../../common/Modal/Modal";
import { rem } from "../../../styles/tools";

import {
  REGISTRATION_DATA_NOTIFICATION_TYPE,
  ACTION_TYPE_PERSON_REGISTRATION_CONFIRM_INFORMATION,
  UPDATE_PERSON_REGISTRATION,
  CONFIRM_PERSONAL_REGISTRATION
} from "../../../utils/constants";
import { formatCPF } from "../../../utils/formatNumber";
import { Button, AlertMessage } from "react-bocombbm-components";
import EFTToken from "../../common/EFTToken";
import ExitConfirmation from "../../common/ExitConfirmation";
import DefaultShimmerLoading from "../../common/DefaultShimmerLoading";
import { redirect } from "../../../utils/redirect";
import checkForSectionFillCompletion, {
  validateAllRegistration
} from "./registrationDataFormValidator";
import FormAgreement from "./FormAgreement";
import { checkIfHasAccess } from "../../common/CanAccess/CanAccess";
import { formatDateToLocale } from "../../../utils/formatDate";
import { scrollToTop } from "../../../utils/dom";

export const StateContext = React.createContext(null);

class RegistrationDataForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      formModel: null,
      shimmerLoding: true,
      agree: true,
      isAuthenticating: false,
      isExiting: false,
      personalDataToken: null,
      loading: false,
      isFilled: {},
      submitUpdatedForm: false,
      temporaryInvalidFields: {},
      closeAll: null,
      canUpdate: null,
      isImpersonating: null
    };

    this.changeFormModel = this.changeFormModel.bind(this);
    this.personalInfomation = this.personalInfomation.bind(this);
    this.submitRegistration = this.submitRegistration.bind(this);
    this.setRegistrationAgreement = this.setRegistrationAgreement.bind(this);
    this.prepareSubmission = this.prepareSubmission.bind(this);
    this.validateSection = this.validateSection.bind(this);
    this.openAuth = this.openAuth.bind(this);
  }

  async componentWillMount() {
    const { getRegistrationFormData, getNotification, userInfo } = this.props;

    await getRegistrationFormData();
    await getNotification();

    this.registrationDataNotification = this.props.notification.find(
      item => item.type === REGISTRATION_DATA_NOTIFICATION_TYPE
    );

    const clonedRegistrationData = deepClone(this.props.registrationFormData);
    clonedRegistrationData.content.documents = {};

    const canUpdate =
      typeof MOBILEAPP_Logout === "function" // eslint-disable-line
        ? __SHOW_EDITABLE_REGISTRATION_CONTENT_MOBILE__ === "true" &&
          checkIfHasAccess(userInfo, [UPDATE_PERSON_REGISTRATION])
        : checkIfHasAccess(userInfo, [UPDATE_PERSON_REGISTRATION]);

    const isImpersonating = !checkIfHasAccess(
      userInfo,
      [UPDATE_PERSON_REGISTRATION, CONFIRM_PERSONAL_REGISTRATION],
      true
    );

    this.setState({
      formModel: clonedRegistrationData,
      shimmerLoding: false,
      canUpdate,
      isImpersonating
    });
  }

  async openAuth(isFormEqual) {
    const {
      postConfirmRegistrationFormData,
      postUpdateRegistrationFormData
    } = this.props;
    this.setState({ loading: true });

    isFormEqual
      ? await postConfirmRegistrationFormData()
      : await postUpdateRegistrationFormData(this.state.formModel.content);

    this.setState({
      personalDataToken: isFormEqual
        ? this.props.dataPersonalRegistration.confirmationToken
        : this.props.dataPersonalRegistration.updateToken,
      submitUpdatedForm: !isFormEqual,
      isAuthenticating: true
    });
  }

  async submitRegistration() {
    const {
      openModal,
      closeModal,
      putConfirmRegistrationFormData,
      putUpdateRegistrationFormData
    } = this.props;
    const { agree } = this.state;

    this.state.submitUpdatedForm
      ? await putUpdateRegistrationFormData()
      : await putConfirmRegistrationFormData(!!agree);

    closeModal();
    openModal({
      type: MODAL_TYPES.PAGE_AS_MODAL,
      width: rem(800),
      children: () => <RegistrationDataFormResult />
    });
  }

  isFormDataEmpty() {
    return !this.props.registrationFormData;
  }

  validateSection(changedInfo, section) {
    const { formModel, canUpdate } = this.state;
    //isFilled is a boolean except for documents that is an object
    const isFilled = canUpdate
      ? checkForSectionFillCompletion(changedInfo, section, formModel.content)
      : true;

    this.setState(prevState => ({
      isFilled: {
        ...prevState.isFilled,
        [section]: isFilled
      },
      closeAll: null
    }));
  }

  changeFormModel(changedInfo, section) {
    this.setState(prevState => ({
      formModel: {
        ...prevState.formModel,
        content: {
          ...prevState.formModel.content,
          [section]: changedInfo
        }
      }
    }));
  }

  setRegistrationAgreement(hasAgreed) {
    this.setState({
      agree: hasAgreed
    });
  }

  personalInfomation(personalRegistrationDetails) {
    return (
      <Fragment>
        <b>{personalRegistrationDetails.name}</b>
        {`, ${translate("ATUCAD_ID_NUMBER")} `}
        <b>{personalRegistrationDetails.documentNumber}</b>
        {personalRegistrationDetails.documentIssuingBody && (
          <Fragment>
            {`, ${translate("ATUCAD_ISSUED_BY")} `}
            <b>{personalRegistrationDetails.documentIssuingBody}</b>
          </Fragment>
        )}
        {personalRegistrationDetails.documentDateOfIssue && (
          <Fragment>
            {`, ${translate("ATUCAD_ON")} `}
            <b>
              {formatDateToLocale(
                personalRegistrationDetails.documentDateOfIssue
              )}
            </b>
          </Fragment>
        )}
        {`, ${translate("ATUCAD_AND_CPF_NUMBER")} `}
        <b>{formatCPF(personalRegistrationDetails.cpf)}.</b>
      </Fragment>
    );
  }

  prepareSubmission() {
    this.setState({
      temporaryInvalidFields: {}
    });

    const { isValidatedBySection, invalidsBySection } = validateAllRegistration(
      this.state.formModel.content
    );

    const validated = Object.values(isValidatedBySection).every(item => {
      if (typeof item === "object") {
        const keys = Object.keys(item);
        return keys.every(key => item[key]);
      }
      return item;
    });

    if (!validated) {
      setTimeout(() => {
        this.props.openToastr({
          text: translate("REQUIRED_FIELDS_OMMITED"),
          error: true,
          timeout: 5000
        });
        this.setState({
          temporaryInvalidFields: invalidsBySection,
          isFilled: isValidatedBySection
        });
        scrollToTop();
      }, 500);
      return this.setState({
        closeAll: true
      });
    }

    this.props.registrationFormData.content.documents = {};

    const equalityFormsCheck =
      JSON.stringify(this.props.registrationFormData) ===
      JSON.stringify(this.state.formModel);

    validated && this.openAuth(equalityFormsCheck);
  }

  renderRegistrationDataForm() {
    const {
      personalRegistrationDetails,
      //taxResidences,
      contacts
      //fatcaInformation,
      //attorneysInFact
    } = this.props.registrationFormData.content;

    const { formModel, agree, canUpdate, isImpersonating } = this.state;
    const fomModelContent = formModel["content"];

    return (
      <Fragment>
        <UserInformation>
          <Greeting>
            {translate("ATUCAD_HELLO")}, {personalRegistrationDetails.name}
          </Greeting>
          <Warning>{translate("ATUCAD_READ_BELOW")}</Warning>
          <BasicInfo>
            {Object.keys(personalRegistrationDetails).length > 0 &&
              this.personalInfomation(personalRegistrationDetails)}
          </BasicInfo>
        </UserInformation>
        <InnerWrapContent>
          {fomModelContent.personalRegistrationDetails && (
            <PersonalDetails
              onChange={this.changeFormModel}
              validateSection={this.validateSection}
              disabled={!canUpdate}
            />
          )}

          {fomModelContent.contacts && (
            <CorrespondenceAndContacts
              originalData={contacts}
              onChange={this.changeFormModel}
              validateSection={this.validateSection}
              disabled={!canUpdate}
            />
          )}

          {/* {(canUpdate || isImpersonating) &&
            fomModelContent.professionalInformation && (
              <ProfessionalInfo
                formModel={fomModelContent}
                onChange={this.changeFormModel}
                validateSection={this.validateSection}
                isFilled={isFilled}
                invalidFields={temporaryInvalidFields.professionalInformation}
                disabled={!canUpdate || isImpersonating}
              />
            )} */}

          {/* {fomModelContent.representationAuthorization !== null && (
            <RepresentationAuthorization
              formModel={fomModelContent}
              originalData={attorneysInFact}
              onChange={this.changeFormModel}
              isFilled={isFilled}
              validateSection={this.validateSection}
              invalidFields={temporaryInvalidFields}
              disabled={!canUpdate}
            />
          )} */}

          {/* {(canUpdate || isImpersonating) &&
            fomModelContent.politicallyExposedPerson && (
              <PoliticallyExposedPerson
                formModel={fomModelContent}
                onChange={this.changeFormModel}
                validateSection={this.validateSection}
                invalidFields={temporaryInvalidFields.politicallyExposedPerson}
                isFilled={isFilled}
                disabled={!canUpdate || isImpersonating}
              />
            )} */}

          {/* {(canUpdate || isImpersonating) &&
            fomModelContent.relatedPersonInformation && (
              <RelatedPerson
                formModel={fomModelContent}
                onChange={this.changeFormModel}
                validateSection={this.validateSection}
                isFilled={isFilled}
                invalidFields={temporaryInvalidFields.relatedPersonInformation}
                disabled={!canUpdate || isImpersonating}
              />
            )} */}

          {/* {fomModelContent.purposeWithTheInstitution && (
            <PurposeWithInstitution
              formModel={fomModelContent}
              onChange={this.changeFormModel}
              isFilled={isFilled}
              invalidFields={temporaryInvalidFields.purposeWithTheInstitution}
              validateSection={this.validateSection}
              disabled={!canUpdate}
            />
          )} */}

          {/* {fomModelContent.taxResidences && (
            <OtherNationality
              formModel={fomModelContent}
              originalData={taxResidences}
              onChange={this.changeFormModel}
              isFilled={isFilled}
              validateSection={this.validateSection}
              invalidFields={temporaryInvalidFields}
              disabled={!canUpdate}
            />
          )} */}

          {/* {(canUpdate || isImpersonating) &&
            fomModelContent.fatcaInformation && (
              <ForeignAccountTaxCA
                formModel={fomModelContent}
                onChange={this.changeFormModel}
                isFilled={isFilled}
                originalData={fatcaInformation}
                invalidFields={temporaryInvalidFields}
                validateSection={this.validateSection}
                disabled={!canUpdate || isImpersonating}
              />
            )} */}

          {fomModelContent.investmentDetails && (
            <FinancialInformation
              onChange={this.changeFormModel}
              validateSection={this.validateSection}
              disabled={!canUpdate}
            />
          )}
        </InnerWrapContent>
        <WrapperAlert>
          <AlertMessage icon="Attention" type="neutral">
            <WrapperText>
              <TextBold>{translate("ATUCAD_ALERT_MESSAGE")}</TextBold>
              <span>{translate("ATUCAD_ALERT_MESSAGE_2")}</span>
            </WrapperText>
          </AlertMessage>
        </WrapperAlert>
        <Understatement>
          {canUpdate ? (
            <Button
              dataTest="registration-data-form-confirm"
              spacing={{ top: "l" }}
              type="conclusive"
              onClick={this.prepareSubmission}
              loading={this.state.loading}
            >
              {translate("ATUCAD_UPDATE_INFORMATION")}
            </Button>
          ) : !isImpersonating ? (
            <FormAgreement
              agree={agree}
              onAgree={this.setRegistrationAgreement}
              onSubmit={this.openAuth}
              loading={this.state.loading}
            />
          ) : null}
        </Understatement>
      </Fragment>
    );
  }

  renderMFA() {
    const params = {
      actionType: ACTION_TYPE_PERSON_REGISTRATION_CONFIRM_INFORMATION,
      payload: {
        id: this.state.personalDataToken,
        informationChanged: !this.state.agree
      }
    };

    return (
      <TokenWrapper>
        <EFTToken
          resendFunc={e => e.preventDefault()}
          onMFAConfirmation={() => this.submitRegistration()}
          createAuthCodeParams={params}
          mfaTokenValidated={this.props.mfaTokenValidated}
          onMFAError={() => {}}
        />
      </TokenWrapper>
    );
  }

  renderExitConfirmation() {
    return (
      <ExitConfirmation
        onClickExit={() => this.handleClose()}
        onClickCancel={() => this.setState({ isExiting: false })}
      />
    );
  }

  handleClose() {
    const { setNotificationStatus, closeModal } = this.props;
    closeModal();

    setNotificationStatus(REGISTRATION_DATA_NOTIFICATION_TYPE);
    redirect("/home");
  }

  render() {
    const { isAuthenticating, isExiting, shimmerLoding } = this.state;
    const { methodChanging } = this.props;

    if (this.isFormDataEmpty()) {
      return null;
    }

    return (
      <StateContext.Provider value={this.state}>
        <Wrapper>
          {!isExiting ? (
            <Header
              title={
                !this.props.methodChanging
                  ? translate("ATUCAD_KEEP_YOUR_REGISTRATION_UP_TO_DATE")
                  : null
              }
              onClickClose={() => this.setState({ isExiting: true })}
              onClickBack={
                methodChanging
                  ? () => this.props.changeFactorTogle(false)
                  : null
              }
              dataTest="registration-data-close"
            />
          ) : (
            <Header
              title={translate("ATUCAD_KEEP_YOUR_REGISTRATION_UP_TO_DATE")}
              dataTest="registration-data-close"
            />
          )}
          {shimmerLoding && (
            <DefaultShimmerLoading repeat={3} innerRepeat={4} />
          )}
          {isExiting && this.renderExitConfirmation()}
          {isAuthenticating && !isExiting && this.renderMFA()}
          {this.state.formModel &&
          Object.keys(this.state.formModel).length > 0 &&
          !isAuthenticating &&
          !isExiting
            ? this.renderRegistrationDataForm()
            : null}
        </Wrapper>
      </StateContext.Provider>
    );
  }
}

RegistrationDataForm.propTypes = {
  registrationFormData: object,
  getRegistrationFormData: func.isRequired
};

export default RegistrationDataForm;
