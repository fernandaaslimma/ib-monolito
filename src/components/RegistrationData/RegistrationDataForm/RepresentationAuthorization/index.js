import RepresentationAuthorization from "./RepresentationAuthorization";

export default RepresentationAuthorization;
