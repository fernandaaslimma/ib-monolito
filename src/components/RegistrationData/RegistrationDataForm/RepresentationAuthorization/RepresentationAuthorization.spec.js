import React from "react";
import { shallow } from "enzyme";
import RepresentationAuthorization from "./RepresentationAuthorization";
import Radio from "../../../common/Radio";
import { IncrementalInput } from "react-bocombbm-components";
import SectionedFormContent from "../../../common/SectionedFormContent";
import { StateContext } from "../RegistrationDataForm";

jest.mock("../../../../utils/fetchHandler");
jest.mock("../../../../utils/i18n", () => ({
  translate: id => id,
  getLanguage: () => "en-US",
  isZhCN: () => false,
  isPtBR: () => false
}));

const props = {
  onChange: jest.fn(),
  validateSection: jest.fn()
};

const state = {
  formModel: {
    content: {
      representationAuthorization: true,
      attorneysInFact: [
        {
          name: "Procurador 1",
          cpf: "11122233344",
          startDate: 1593117908000,
          endDate: 1593117908000
        },
        {
          name: "Procurador 2",
          cpf: "55566677788",
          startDate: 1593117908000,
          endDate: null
        }
      ]
    }
  },
  isFilled: {
    attorneysInFact: false
  },
  temporaryInvalidFields: {}
};

const falseState = {
  formModel: {
    content: {
      representationAuthorization: false,
      attorneysInFact: []
    }
  }
};

describe("RepresentationAuthorization component", () => {
  it("should match snapshot with props", () => {
    expect(
      shallow(
        <StateContext.Provider value={state}>
          <RepresentationAuthorization {...props} />
        </StateContext.Provider>
      )
    ).toMatchSnapshot();
  });

  it("should match snapshot with false representation", () => {
    React.useContext = jest.fn(() => falseState);
    expect(
      shallow(
        <StateContext.Provider value={state}>
          <RepresentationAuthorization {...props} />
        </StateContext.Provider>
      )
    ).toMatchSnapshot();
  });

  it.skip("should call onChange when change radio value", () => {
    const wrapper = shallow(
      <StateContext.Provider value={state}>
        <RepresentationAuthorization {...props} />
      </StateContext.Provider>
    );
    wrapper
      .find(Radio)
      .at(1)
      .prop("onChange")({ target: { value: "false" } });
    expect(props.onChange).toHaveBeenCalledTimes(2);
  });

  it.skip("should call onChange when change radio value to true", () => {
    const wrapper = shallow(
      <StateContext.Provider value={state}>
        <RepresentationAuthorization {...props} />
      </StateContext.Provider>
    );
    wrapper
      .find(Radio)
      .at(0)
      .prop("onChange")({ target: { value: "true" } });
    expect(props.onChange).toHaveBeenCalled();
  });

  it.skip("should validate an incremental field", () => {
    const wrapper = shallow(
      <StateContext.Provider value={state}>
        <RepresentationAuthorization {...props} />
      </StateContext.Provider>
    );
    expect(
      wrapper
        .find(IncrementalInput)
        .prop("inputs")[0]
        .valid("name", 0)
    ).toBeTruthy();
  });

  it.skip("should change an incremental field", () => {
    const wrapper = shallow(
      <StateContext.Provider value={state}>
        <RepresentationAuthorization {...props} />
      </StateContext.Provider>
    );
    wrapper.find(IncrementalInput).prop("onChange")([]);
    expect(props.onChange).toHaveBeenCalledWith([], "attorneysInFact");
  });

  it.skip("should validate section when toggle section container show/hide", () => {
    const wrapper = shallow(
      <StateContext.Provider value={state}>
        <RepresentationAuthorization {...props} />
      </StateContext.Provider>
    );
    const fileInput = wrapper.find(SectionedFormContent);

    fileInput.prop("callback")();
    expect(props.validateSection).toHaveBeenCalled();
  });

  it("should display AlertMessage", () => {
    state.temporaryInvalidFields = {
      attorneysInFact: ["name", "cpf", "startDate", "endDate"]
    };
    const wrapper = shallow(
      <StateContext.Provider value={state}>
        <RepresentationAuthorization {...props} />
      </StateContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
