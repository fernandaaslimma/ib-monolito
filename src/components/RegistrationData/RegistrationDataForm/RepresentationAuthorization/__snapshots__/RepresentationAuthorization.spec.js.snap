// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepresentationAuthorization component should display AlertMessage 1`] = `
<RepresentationAuthorization
  onChange={[MockFunction]}
  validateSection={[MockFunction]}
/>
`;

exports[`RepresentationAuthorization component should match snapshot with false representation 1`] = `
<RepresentationAuthorization
  onChange={[MockFunction]}
  validateSection={[MockFunction]}
/>
`;

exports[`RepresentationAuthorization component should match snapshot with props 1`] = `
<RepresentationAuthorization
  onChange={[MockFunction]}
  validateSection={[MockFunction]}
/>
`;
